### ISSUE TITLE
Eliminação recursiva

### ISSUE THREAD

**Issue by beatrizRamos** [davidalain/SuperEinstein#17]
No método de deletar uma disciplina, tenho que obrigatoriamente chamar o método de deletar assuntos?

@davidalain 
@JMDRibeiro 
@vanessaf5 
@beatrizRamos

**Comment by vanessaf5** [comment]
Sim, pois os assuntos não podem existir sem a disciplina. Os assuntos são por disciplina.

**Comment by JMDRibeiro** [comment]
Mas aí, minha dúvida e a de @beatrizRamos, é se **no método deletar da classe DisciplinaRepository** - por exemplo - **deveremos chamar o método deletar de todos os assuntos que tem o id da disciplina**. 
Se isso for feito, como consequência, teremos de deletar também todas as questões relacionadas aos assuntos ligados à essa disciplina e, por conseguinte, todas as alternativas que tiverem o id dessa questão.
Contudo, **não sei se isso é adequado**- o método de um Repository invocar vários outras funções de vários outros Repository's.
No que tange ao banco parece ser o lógico a se fazer, mas, não sei qual a repercussão disso na Orientação a Objetos.
Por isso a dúvida.

**Comment by davidalain** [comment]
Olhem isto:

https://stackoverflow.com/questions/30118599/having-a-repository-dependent-on-another-repository

Na utilização de arquitetura em camadas Isto é resolvido na camada acima do
repositório. Esta camada pode chamar qualquer repositório, mas um
repositório não pode chamar o outro. O repositório só faz requisições sobre
a entidade que ele trata. Por, exemplo, o repositório Disciplina só acessa
a tabela disciplina.
Então deverá ser criada uma camada chamada *Service *que tem as regras do
que pode ou não ser feito no sistema (regras de negócio). E a camada
*Repository
*só faz acessos simples ao banco.

Isto vale pra o app Android e WebService.

Recomendo seguir este padrão na implementação do repositório:
https://medium.com/@krzychukosobudzki/repository-design-pattern-bc490b256006

Isto irá ajudar vocês sobre como fazer a camada de serviço (onde ficam as
regras de negócio):
https://www.devmedia.com.br/construindo-projetos-em-camadas-usando-java/6418

Em qui, 3 de mai de 2018 18:10, JMDRibeiro <notifications@github.com>
escreveu:
